beacon-chain/blockchain/receive_block.go

```diff
@@ -209,3 +210,11 @@ func isCompetingBlock(root []byte, slot uint64, headRoot []byte, headSlot uint64
 		competingBlks.Inc()
 	}
 }
+
+//ParentExists checks block parent is in store.
+func (s *Service) ParentExists(ctx context.Context, block *ethpb.BeaconBlock) bool {
```

Comment:
You dont need this, InitialSync should have access to the DB, they can just call it there. I don't think BlockReceive should provide this just to call DB, It's not chain service dependent

Reply:
you are right. moving it